### PR TITLE
Add LenSqr functions

### PR DESCRIPTION
**Pull request by Zyl9393**
Added SqrLen() functions which return squared vector length. This can make some lines shorter when variable names are long, e.g.:
`worldSpaceFragmentCenter.Dot(worldSpaceFragmentCenter)`
vs.
`worldSpaceFragmentCenter.SqrLen()`
Understandable if not wanted, though. Mostly learning how 2 github over here. Have a nice day.

### CODE REVIEW

File: mgl64/matstack/transformstack_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// This file is generated from mgl32/matstack/transformstack_test.go; DO NOT EDIT
+// This file is generated from mgl32/matstack\transformstack_test.go; DO NOT EDIT
```
Comment (pwaller) [inline mgl64/matstack/transformstack_test.go]: Hmm, I take it this was generated on Windows. Oh dear. Probably need a bugfix to `codegen.go`. Can you amend the commit to remove these?

Comment (Zyl9393) [reply mgl64/matstack/transformstack_test.go]: > Probably need a bugfix to `codegen.go`.

Made #80

**Comment by pwaller** [comment]
Thanks for fixing the issues I raised!

One thing that occured to me as I go to merge it:

How about `LenSquared` as the name of the function? Then if you're using an editor with autocomplete, it will show up as you're looking for `Len` - improving the discoverability. Also it would appear next to `Len` in the documentation: https://godoc.org/github.com/go-gl/mathgl/mgl32

Let me know what you think. Happy to accept it as is otherwise - though I will squash the history down to just "Add ... functions", to avoid the merge commit going from master into the development branch.

**Comment by MMulthaupt** [comment]
I would expect autocomplete to list the function regardless of where in the name the substring is found, but regarding the documentation, you raise a good point. Maybe call it `LenSqr` then, or `LenSq` even? Having it short was one of main reasons for making this, and `Sq` is a well understood shortcut in anything math-related; even Java refrains from its usual verbosity, [calling the square root function in its Math package `sqrt`](https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html#sqrt(double)).

**Comment by pwaller** [comment]
OK, then :+1: to `LenSqr` from me.

**Comment by Zyl9393** [comment]
Should be good now.

**Pull request merged**